hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvokeOnPartitions.java

```diff
@@ -124,7 +126,7 @@ private void retryFailedPartitions() throws InterruptedException, ExecutionExcep
 
         for (Integer failedPartition : failedPartitions) {
             Operation operation;
-            if (operationFactory instanceof PartitionAwareOperationFactory) {
+            if (isPartitionAwareFactory(operationFactory)) {
```

Comment:
Yes good point, here my goal is to fix this issue: hazelcast/hazelcast-enterprise#1213 on what we have already and porting it to maintenance. `PartitionAwareOperationFactory` and `OperationFactory` relation can be handled in a separate issue. Does it make sense?